.github/workflows/scripts/fetch_packages_metadata.py

```diff
@@ -0,0 +1,75 @@
+from requests import get
+from datetime import datetime, timedelta
+from concurrent.futures import ThreadPoolExecutor
+from itertools import islice, chain
+import sys
+import subprocess
+
+
+def get_info_from_pypi(pkg):
+    url = f"https://pypi.org/pypi/{pkg}/json"
+    try:
+        res = get(url).json()
+        return res
+    except Exception as e:
+        print(f"request to pypi failed for <{pkg}>: {e}")
+        return None
+
+
+def get_outdated_pkg_list():
+    out = subprocess.check_output(["python", "-m", "pip", "list", "--outdated", "--format", "freeze"])
```

Comment:
If we use `--outdated` then we won't process any packages that are up-to-date, which is a problem. The [issue that motivated this enhancement request](https://github.com/elyra-ai/elyra/issues/2778) is a good example for why we cannot ignore these packages. What happened was that in test build XXX package version x.y.z was installed and everything worked. The following day (test build XXX+1) package version x.y.z+1 was installed and the test failed. We try to avoid capping versions and it's therefore common that two test builds use different dependency versions, even though nothing has changed in the Elyra code.

Reply:
changed to pip freeze